app/liquid_tags/user_tag.rb

```diff
@@ -20,12 +24,16 @@ def render(_context)
     )
   end
 
-  def parse_username_to_user(user)
-    user = User.find_by(username: user)
-    raise StandardError, "Invalid username" if user.nil?
+  def parse_username_to_user(username)
+    user = User.find_by(username: username)
+    return username if user.nil?
```

Comment:
I'm not a big fan of returning two different types here and then doing a type check in `user_not_found?`. If you return some sort of "null object" here the code would be a lot nicer IMHO.

```ruby
OpenStruct.new(username: username, 'persisted?': false)
```

With that we could get rid of `user_not_found?` and update `initialize` to use something like

```ruby
return unless user.persisted?
```

and it may also need some work in `render`.

Maybe we don't even need the actual username though and could replace it with something like "deleted user". Then we could even move the null object to something like `Constants::DeletedUser` or `Users::DeletedUser` and re-use it every time we need to deal with something like this, which often greatly reduces the amount of conditionals in an app. WDYT? /cc @rhymes @benhalpern